python/ray/tune/ray_trial_executor.py

```diff
@@ -355,24 +355,23 @@ def _update_avail_resources(self, num_retries=5):
         for i in range(num_retries):
             try:
                 resources = ray.global_state.cluster_resources()
+                if "CPU" not in resources:
+                    resources["CPU"] = 0
             except Exception:
                 # TODO(rliaw): Remove this when local mode is fixed.
                 # https://github.com/ray-project/ray/issues/4147
                 logger.debug("Using resources for local machine.")
                 resources = ray.services.check_and_update_resources(
                     None, None, None)
-            if not resources:
```

Comment:
@richardliaw We're changing resource semantics - zero capacity resources are now not included in resource datastructures. For instance instead of returning {resource: 0} we now return an empty dictionary.

To make tests work, I've made some changes to resource handling in tune - can you please check if this looks okay?

Reply:
I pushed a small tweak - thanks for letting me know!